Sources/OutlineView/OutlineViewDragAndDrop.swift

```diff
@@ -0,0 +1,148 @@
+import Foundation
+import AppKit
+
+
+/// A protocol for use with `OutlineView`, implemented by an object (most likely
+/// the data source) to handle receiving items that are dragged into the `OutlineView`.
+@available(macOS 10.15, *)
+public protocol DropReceiver {
+    associatedtype DataElement: Identifiable
+    
+    /// An array of `PasteboardType`s that the `DropReceiver` is able to read
+    var acceptedTypes: [NSPasteboard.PasteboardType] { get }
```

Comment:
A very minor concern that having `acceptedTypes` be a part of the protocol may suggest that the value can be changed later (for drops triggered after the change). That is of course not supported as the `acceptedTypes` are set at initialization of the outline view controller.

But that might just be me overthinking it.

Reply:
@ZofiaBernadetta , I had a thought that it could be acceptable to have `acceptedTypes` change during runtime, although I can't think of what the circumstance would be. Either way, I think keeping `acceptedTypes` in the protocol is preferable. Otherwise an `OutlineView` that accepts drops would need a `DropReceiver` and an array of `acceptedTypes` declared in its initializer (unless I'm missing another option). Happy to discuss further, though.

Reply:
> Either way, I think keeping acceptedTypes in the protocol is preferable.

I do agree that it makes for a better API. I suppose SwiftUI gets around this by letting you choose on a per drop basis
https://developer.apple.com/documentation/swiftui/view/ondrop(of:istargeted:perform:)-f15m

Perhaps we support attaching a drop receiver using a modifier?

```swift
@available(macOS 10.15, *)
public extension OutlineView {
    func onDrop(of supportedPasteboardTypes: [NSPasteboard.PasteboardType], receiver: Drop) -> Self {
        var mutableSelf = self
        mutableSelf.dropReceiver = receiver
        return mutableSelf
    }
}
```

You can remove the `dropReceiver` parameter from the normal init and make the `dropReceiver` an optional property on `OutlineView` so that it's inferred to be `nil`. You will need a duplicate of the same init in an extension with `Drop == NoDropReceiver<Data.Element>` as a constraint (which you already have!).

the NoDropReceiver could even be implemented as an uninhabited enum as you would not need to ever initialize it. 
```swift
@available(macOS 10.15, *)
public enum NoDropReceiver<Element: Identifiable>: DropReceiver {
    public typealias DataElement = Element
    public var acceptedTypes: [NSPasteboard.PasteboardType] { fatalError() }
    
    public func readPasteboard(item: NSPasteboardItem) -> DraggedItem<Element>? {
        fatalError()
    }
    
    public func validateDrop(target: DropTarget<Element>) -> ValidationResult<Element> {
        fatalError()
    }
    
    public func acceptDrop(target: DropTarget<Element>) -> Bool {
        fatalError()
    }
}
```

When using `.onDrop(of: [], receiver: ...)`, the type of the OutlineView will automatically be inferred to have the correct DropReceiver.

What do you think about it?

Reply:
Very cool trick! I'm testing it out now... Too bad it doesn't cut down the number of initializers we have with the OutlineView, but otherwise it's quite nice. And keeps things similar to the other SwiftUI onDrop modifier. I'll work on this now and see what I can do.

Reply:
Works great, and feels much cleaner. https://github.com/Sameesunkaria/OutlineView/pull/10/commits/6a32aa9e9615122d330b17558ad01d91fa2535b5

I didn't separate the `acceptedTypes` from the `DropReceiver` protocol, so the modifier just looks like `onDrop(receiver:)` rather than the SwiftUI List version (`onDrop(of:delegate:)`). Maybe that could be changed if you like. Otherwise, I think it's still a good improvement!

Reply:
Unless there is a reason for us to deviate from the SwiftUI API, we should stick to providing a similar API. This helps people migrating their existing SwiftUI code, as they are already familiar with what SwiftUI has to offers.

This discussion thread started with a concern around including `acceptedTypes` in the delegate, and now we have a perfect way to address that.

Side note: It would be nice to add support for uniform type identifiers, if they are trivial to add. Not saying that you should work on it, or if you do work on it, that it be included in this PR. Everything we do here must be seen through the glasses of a "temporary workaround" that is intended to be removed once Apple fixed their stuff, so if adding support for uniform type identifiers does not materially benefit the users, then there is no point in adding it. I was only thinking about it in terms of the completeness of the API.

Reply:
Sounds good. I'll work on making that change today or tomorrow, which shouldn't be too difficult. As for using UTTypes, I'll have to do a little research there. I know the `NSOutlineView` uses `NSPasteboard.PasteboardType` rather than UTTypes, while SwiftUI's `onDrop` use UTTypes. I haven't checked if there's a direct relation between those types. I'll get back to you on that.

Reply:
Moved `acceptedTypes` in this commit: https://github.com/Sameesunkaria/OutlineView/pull/10/commits/df3187a72a79d89a9d62d030b41b553b8a820596

It is probably possible to use UTTypes rather than NSPasteboard.PasteboardType, but I'm not sure it would be worth the trouble. I think we would need to be able to translate both types to and from each other so that we could pass UTTypes to NSOutlineViewDataSource methods as PasteboardTypes, and then back from the data source methods to UTTypes going back to the DropReceiver. And we'd have to change the output type for `dragDataSource`, since it currently outputs NSPasteboardItems, which also use PasteboardTypes. I'd be afraid that doing all that translating back and forth would just introduce more room for errors :-)

Reply:
Thanks for looking into it.